Core/src/org/sleuthkit/autopsy/configurelogicalimager/LogicalImagerRule.java

```diff
@@ -0,0 +1,254 @@
+    ) {
+        this.shouldAlert = shouldAlert;
+        this.shouldSave = shouldSave;
+        this.name = name;
+        this.description = description;
+        this.extensions = extensions;
+        this.filenames = filenames;
+        this.paths = paths;
+        this.fullPaths = fullPaths;
+        
+        this.sizeRange.put("min", minFileSize); // NON-NLS
+        this.minFileSize = minFileSize;
+        this.sizeRange.put("max", maxFileSize); // NON-NLS
+        this.maxFileSize = maxFileSize;
+        this.dateRange.put("min-days", minDays); // NON-NLS
+        this.minDays = minDays;
+        this.dateRange.put("min-date", minDate); // NON-NLS
+        this.minDate = minDate;
+        this.dateRange.put("max-date", maxDate); // NON-NLS
+        this.maxDate = maxDate;
+    }
+
+    public LogicalImagerRule() {
+        this.shouldAlert = false; // default
+        this.shouldSave = true; // default
+        this.description = null;
+        this.name = null;
+    }
+    
+    public Boolean isShouldAlert() {
+        return shouldAlert;
+    }
+
+    public Boolean isShouldSave() {
+        return shouldSave;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public List<String> getExtensions() {
+        return extensions;
+    }
+
+    public List<String> getFilenames() {
+        return filenames;
+    }
+
+    public List<String> getPaths() {
+        return paths;
+    }
+
+    public List<String> getFullPaths() {
+        return fullPaths;
+    }
+
+    public Integer getMinFileSize() {
+        return minFileSize;
+    }
+
+    public Integer getMaxFileSize() {
+        return maxFileSize;
+    }
+
+    public Integer getMinDays() {
+        return minDays;
+    }
+
+    public Integer getMinDate() {
+        return minDate;
+    }
+
+    public Integer getMaxDate() {
+        return maxDate;
+    }
+
+    /**
+     * Builder class
+     */
+    public static class Builder {
+        private Boolean shouldAlert = null;
+        private Boolean shouldSave = null;
+        private String name = null;
+        private String description = null;
+        private List<String> extensions = null;
+        private List<String> filenames = null;        
+        private List<String> paths = null;        
+        private List<String> fullPaths = null;        
+        private Integer minFileSize = null;
+        private Integer maxFileSize = null;
+        private Integer minDays = null;
+        private Integer minDate = null;
+        private Integer maxDate = null;
+        
+        public Builder shouldAlert(boolean shouldAlert) {
+            this.shouldAlert = shouldAlert;
+            return this;
+        }
+        
+        public Builder shouldSave(boolean shouldSave) {
+            this.shouldSave = shouldSave;
+            return this;
+        }
+        
+        public Builder name(String name) {
```

Comment:
I agree with codacy that it is odd that this method and others have the same name as your field. Is there a reason not to name them "setName()" etc?

Reply:
I don't agree. At least in the platforms and clouds code, that is how we do builders.
Just extra typing :(